### PR TITLE
Create spellbook for latest bitcoin balance

### PR DESCRIPTION
**Pull request by gandalftheblockchain**
A follow up to @longcnttbkhn PR https://github.com/duneanalytics/spellbook/pull/4195 and getting latest balance from balances_bitcoin_satoshi_day table. 

Great work @longcnttbkhn! on balances_bitcoin_satoshi_day.sql 
I'd like to add updated_at to this model to provide context on when the model last ran, let me know your thoughts

### CODE REVIEW

**Comment by gandalftheblockchain** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by gandalftheblockchain** [comment]
recheck

File: models/balances/bitcoin/latest_balances_bitcoin_sataoshi_day.sql
Comment (jeff-dude) [inline models/balances/bitcoin/latest_balances_bitcoin_sataoshi_day.sql:1]: please rename file to
`balances_bitcoin_satoshi_latest_day.sql`

file name should always match schema / table alias in the config block

File: models/balances/bitcoin/balances_bitcoin_schema.yml [hunk 1]
```diff
@@ -21,16 +21,45 @@ models:
...
         name: amount_usd
         description: "USD balance of the wallet"
-      - name: total_asset
+      - &total_asset
+        name: total_asset
         description: "Total asset by USD of the wallet"
+      - &updated_at
+        name: updated_at
+        description: "When the balances were last updated"
+  - name: latest_balances_bitcoin_satoshi_day
```
Comment (jeff-dude) [inline models/balances/bitcoin/balances_bitcoin_schema.yml:45]: be sure to update name here too after change

File: models/balances/bitcoin/latest_balances_bitcoin_sataoshi_day.sql [hunk 2]
```diff
@@ -0,0 +1,41 @@
+{{ config(
+        schema = 'balances_bitcoin',
+        alias = 'satoshi_latest_day',
+        materialized='table',
```
Comment (jeff-dude) [inline models/balances/bitcoin/latest_balances_bitcoin_sataoshi_day.sql:4]: with the upstream spell `balances_bitcoin_satoshi_day` being a view, materializing this as a table is slow and crashing the cluster. i wonder if we need to revisit that other spell to materialize as well, if the plan is to build on top of it.

Comment (jeff-dude) [reply models/balances/bitcoin/latest_balances_bitcoin_sataoshi_day.sql:4]: the issue with `balances_bitcoin_satoshi_day` being made incremental came down to the CTE which builds a sequence over time, then obtaining values to match those dates. it wasn't straightforward to build.

Comment (jeff-dude) [reply models/balances/bitcoin/latest_balances_bitcoin_sataoshi_day.sql:4]: we have an ongoing effort to build transfers / balances on evm blockchains, where the approach may make sense to mimic for bitcoin.

you can find the new approaches here:
https://github.com/duneanalytics/spellbook/tree/main/models/_sector/tokens/ethereum

**Review by jeff-dude — CHANGES_REQUESTED** [review]

File: models/balances/bitcoin/latest_balances_bitcoin_sataoshi_day.sql (hunk 2 above)
Comment (gandalftheblockchain) [reply models/balances/bitcoin/latest_balances_bitcoin_sataoshi_day.sql]: Ok I will keep an eye out on this new approach. seems like we will need to refactor this later on

**Comment by gandalftheblockchain** [comment]
@jeff-dude - trying to check the output of the model by running this command,`select * from test_schema.git_dunesql_16098b71_balances_bitcoin_sataoshi_latest_day`
but getting this error. 
Screenshot 2023-11-16 at 5 30 52 AM
how can i run the output?

**Comment by jeff-dude** [comment]
> @jeff-dude - trying to check the output of the model by running this command,`select * from test_schema.git_dunesql_16098b71_balances_bitcoin_sataoshi_latest_day` but getting this error. Screenshot 2023-11-16 at 5 30 52 AM how can i run the output?

it may have been just bad timing. we run a job 1x/day that cleans up all CI test tables, and may have been right as you built.

File: models/balances/bitcoin/balances_bitcoin_sataoshi_latest_day.sql [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    blockchain
+    , day
+    , wallet_address
+    , amount_raw
+    , amount
+    , price_btc
+    , profit
+    , amount_usd
+    , total_asset
+    , now() as updated_at
```
Comment (jeff-dude) [inline models/balances/bitcoin/balances_bitcoin_sataoshi_latest_day.sql:39]: this likely won't help you on either spell, as it's a view and will calculate `now()` every time you execute a query on it in the app

Comment (jeff-dude) [reply models/balances/bitcoin/balances_bitcoin_sataoshi_latest_day.sql:39]: we can leave as-is for now, then if you revisit to redesign these spells later, it'll be there as a reminder to include

**Comment by jeff-dude** [comment]
i was able to query the latest CI view built in the logs, but returning results is v slow. this whole dbt lineage will likely need refactored to work as incremental models, so you are querying a table rather than view on full history of source tables.

i'm happy to still merge as-is, let me know

**Comment by soochoe** [comment]
@jeff-dude lets merge as is and I will create a subsequent refactoring PR

File: models/balances/bitcoin/balances_bitcoin_sataoshi_latest_day.sql (hunk 3 above)
Comment (soochoe) [reply models/balances/bitcoin/balances_bitcoin_sataoshi_latest_day.sql:39]: Thanks for pointing that out! Yes, let’s do that and I will refactor this later

**Review by jeff-dude — APPROVED** [review]

**Comment by longcnttbkhn** [comment]
Great awesome @gandalftheblockchain

**Pull request merged**